### PR TITLE
Expose metrics through local http page

### PR DESCRIPTION
**Pull request by fokot**
https://github.com/zio/zio-zmx/issues/187

I expose `PMetric` as json

@atooni @adamgfraser

### CODE REVIEW

**Comment by softinio** [comment]
@fokot LGTM ... are there any tests you can add?

**Comment by fokot** [comment]
@softinio I added tests

**Comment by softinio** [comment]
Linting errors for you to resolve, guess run `sbt fmt` and commit and push:

https://shared.softinio.com/Screen-Shot-2021-02-28-14-40-30.97.png#b8fa9f7f8e970246a3c2ed6fcab241d6e4a595bc3e62dec24cbd93ca3702c004

**Comment by atooni** [comment]
Hi, LGTM. Just some observations:

From my understanding you wanted to get rid of the sealed traits in PMetric alltogether and replace them by the case classes. Then your JSON would look like you would expect it to. Perhaps change the test to inspect the expected output and change the code so that the test succeeds. 

I had the same problem that CI yelled at me with Linting errors. I use NVIM or VSC as IDE and have the option "Format on save" enabled and never had a linting error afterwards. Aletrnatively you can run `sbt check` before your commit and see if that comes up with errors. `sbt prepare` would apply formatting to all your scala files.

**Comment by softinio** [comment]
@atooni think `sbt prepare` is specific to the `zio` core repo. Here its just `sbt fmt` as mentioned in my earlier comment to autoformat linting errors FYI 

cc @fokot

**Comment by atooni** [comment]
@softinio You are right of course - in zmx you can do `sbt fmt`and `sbt check` only - just double checked.

**Comment by adamgfraser** [comment]
@fokot What do we need to get this over the finish line?

**Comment by fokot** [comment]
@adamgfraser I need to change all traits to case classes to it is nicer in json

File: build.sbt [hunk 1]
```diff
@@ -38,7 +38,8 @@ libraryDependencies ++= Seq(
   "dev.zio"      %% "zio-nio"      % "1.0.0-RC9",
   "dev.zio"      %% "zio-test"     % zioVersion % "test",
   "dev.zio"      %% "zio-test-sbt" % zioVersion % "test",
-  "org.polynote" %% "uzhttp"       % "0.2.6"    % "test"
+  "org.polynote" %% "uzhttp"       % "0.2.6"    % "test",
+  "dev.zio"      %% "zio-json"     % "0.1"
```
Comment (adamgfraser) [inline build.sbt:42]: I think we probably don't want to add a `zio-json` dependency here. In any other project would be fine but we are really trying to say this is zero dependency other than ZIO. We even want to rip `zio-nio` out. Can we write our own encoders for these data types? I think it should be a very simple hierarchy.

Comment (fokot) [reply build.sbt:42]: Sure I can do it. It is just more error prone, but probably fine. I can add `zio-json` or `circe` to test dependencies and compare results from manual and derived encoding.

Comment (adamgfraser) [reply build.sbt:42]: That's a good idea to compare to `zio-json` in the tests.

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**